vllm/model_executor/layers/spec_decode_base_sampler.py

```diff
@@ -36,9 +36,12 @@ def __init__(self,
         self.num_emitted_tokens: Optional[torch.Tensor] = None
         self.num_draft_tokens: int = 0
 
-    def init_gpu_tensors(self, rank: int) -> None:
+    def init_gpu_tensors(self, device: Union[int, str]) -> None:
         assert self.num_accepted_tokens is None
-        device = f"cuda:{rank}"
+        if isinstance(device, int):
+            device = f"cuda:{device}"
+        elif not isinstance(device, str):
+            raise ValueError(f"Device must be int or str, get {type(device)}")
```

Comment:
nit: move this independent logic to own helper function

Reply:
will leave it as is for now. People can refactor later if they reuse this part of code.